src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,34 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodoId: number,
+  selectedTodo: Todo | null,
+  setSelectedTodoId: (id: number) => void,
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodoId,
+  selectedTodo,
+  setSelectedTodoId,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId)
+        .then(setUser);
+    }
+  }, []);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!selectedTodo || !user ? (
```

Comment:
You should have `isLoading` state for handling loading(like in `App` component). What if you have empty response? You will see loader all the time